xtc/src/common_types.rs

```diff
@@ -0,0 +1,142 @@
+        self.index = index;
+        self
+    }
+}
+
+impl TryFrom<Transaction> for TxRecord {
+    type Error = ();
+    fn try_from(transaction: Transaction) -> Result<TxRecord, ()> {
+        match transaction.kind {
+            TransactionKind::Approve { from, to } => Ok(TxRecord::new(
+                None,
+                from,
+                to,
+                Nat::from(transaction.cycles),
+                Nat::from(transaction.fee),
+                Operation::approve,
+                Int::from(transaction.timestamp),
+                Nat::from(0),
+            )),
+            TransactionKind::Transfer { from, to } => Ok(TxRecord::new(
+                None,
+                from,
+                to,
+                Nat::from(transaction.cycles),
+                Nat::from(transaction.fee),
+                Operation::transfer,
+                Int::from(transaction.timestamp),
+                Nat::from(0),
+            )),
+            TransactionKind::TransferFrom { caller, from, to } => Ok(TxRecord::new(
+                Some(caller),
+                from,
+                to,
+                Nat::from(transaction.cycles),
+                Nat::from(transaction.fee),
+                Operation::transferFrom,
+                Int::from(transaction.timestamp),
+                Nat::from(0),
+            )),
+            TransactionKind::Mint { to } => Ok(TxRecord::new(
+                None,
+                to,
+                to,
+                Nat::from(transaction.cycles),
+                Nat::from(transaction.fee),
+                Operation::mint,
+                Int::from(transaction.timestamp),
+                Nat::from(0),
+            )),
+            TransactionKind::Burn { from, to } => Ok(TxRecord::new(
+                None,
+                from,
+                to,
+                Nat::from(transaction.cycles),
+                Nat::from(transaction.fee),
+                Operation::burn,
+                Int::from(transaction.timestamp),
+                Nat::from(0),
+            )),
+            TransactionKind::CanisterCalled {
+                from,
+                canister,
+                method_name,
+            } => Ok(TxRecord::new(
+                None,
+                from,
+                canister,
+                Nat::from(transaction.cycles),
+                Nat::from(transaction.fee),
+                Operation::canisterCalled,
+                Int::from(transaction.timestamp),
+                Nat::from(0),
+            )),
+            TransactionKind::CanisterCreated { from, canister } => Ok(TxRecord::new(
+                None,
+                from,
+                canister,
+                Nat::from(transaction.cycles),
+                Nat::from(transaction.fee),
+                Operation::canisterCreated,
+                Int::from(transaction.timestamp),
+                Nat::from(0),
+            )),
+            _ => Err(()),
```

Comment:
Seems unreachable to me

Reply:
guess left it from the time where we did have some not handled cases...